### DISCUSSION TITLE
dependencies unmet

### DISCUSSION THREAD

**Discussion by HuntorWorks** [abraunegg/onedrive#1378] [Q&A] [answered]
trying to get onedrive installed on my laptop - i am a new linux user so possible im doing something wrong but using the information here https://www.linuxuprising.com/2020/02/how-to-keep-onedrive-in-sync-with.html on ubuntu 20.04 im getting this error when following the install instructions using ppa
```
sudo apt install onedrive
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies.
 onedrive : Depends: libphobos2-ldc-shared90 (>= 1:1.20.1) but it is not installable
E: Unable to correct problems, you have held broken packages.
```

**Comment by HuntorWorks** [comment]
> trying to get onedrive installed on my laptop - i am a new linux user so possible im doing something wrong but using the information here https://www.linuxuprising.com/2020/02/how-to-keep-onedrive-in-sync-with.html on ubuntu 20.04 im getting this error when following the install instructions using ppa
> ```
> sudo apt install onedrive
> Reading package lists... Done
> Building dependency tree
> Reading state information... Done
> Some packages could not be installed. This may mean that you have
> requested an impossible situation or if you are using the unstable
> distribution that some required packages have not yet been created
> or been moved out of Incoming.
> The following information may help to resolve the situation:
> 
> The following packages have unmet dependencies.
> onedrive : Depends: libphobos2-ldc-shared90 (>= 1:1.20.1) but it is not installable
> E: Unable to correct problems, you have held broken packages.
>```

Seemed to have fixed the issue - turns out I had the checkboxes under unbuntu software and updates > ubuntu software on default selections - since have checked all the top 4 boxes and it works as intended.

**Comment by abraunegg** [comment, marked as answer]
@BeerHuntor 
Glad you solved your issue, however to install the client on Ubuntu you really *should* be following these instructions and not a 3rd party website: https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md

For Ubuntu 20.x you have 2 options:
* Install via the OpenSuSE Repository following these instructions: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md (recommended method)
* Compile via using the correct dependencies: https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#dependencies-ubuntu-18x-ubuntu-19x-ubuntu-20x--debian-9-debian-10---x86_64


Additionally in future, please follow the correct support process as outlined here:
* Have a question - https://github.com/abraunegg/onedrive#have-a-question
* Reporting an issue or bug - https://github.com/abraunegg/onedrive#reporting-an-issue-or-bug

**Comment by yezhengli-Mr9** [comment]
> For Ubuntu 20.x you have 2 options:
> 
> * * Install via the OpenSuSE Repository following these instructions: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md (recommended method)
> * Compile via using the correct dependencies: https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#dependencies-ubuntu-18x-ubuntu-19x-ubuntu-20x--debian-9-debian-10---x86_64

For Ubuntu 20.04, not for sure I got the correct instruction to install. I should not follow third-party instructions which result in the error mentioned:
```java
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 onedrive : Depends: libphobos2-ldc-shared94 (>= 1:1.24.0) but it is not installable
E: Unable to correct problems, you have held broken packages.
```
Best and thanks,

**Comment by abraunegg** [comment]
@yezhengli-Mr9 

> For Ubuntu 20.04, not for sure I got the correct instruction to install. I should not follow third-party instructions which result in the error mentioned:
> `libphobos2-ldc-shared94 (>= 1:1.24.0) but it is not installable`

You are following the incorrect instructions. Please read https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md more carefully and make zero assumptions.

**Reply by roarinpenguin** [reply]
I have followed exactly the instructions at your github link for Ubuntu 22.04 and got the error:
The following packages have unmet dependencies:
 onedrive : Depends: libphobos2-ldc-shared98 (>= 1:1.28.0) but it is not installable
E: Unable to correct problems, you have held broken packages.
Do you have any advice to fix it? Thanks.

**Reply by abraunegg** [reply]
@roarinpenguin 
> Do you have any advice to fix it? Thanks.

Unfortunately you will not have followed the instructions correctly .. you are following the instructions for Ubuntu 22.04 but you are *not* running Ubuntu 22.04 .....

Please re-read the instructions and determine what platform you are running, and follow the *correct* instructions for that Ubuntu version

**Reply by roarinpenguin** [reply]
That is what /etc/lsb-release says...
```
# more /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.4 LTS"
```
Anyways downloading the deb packages from Ubuntu website and installing them with dpkg -i <package name> fixed the installation:
- first you have to install [libllvm11_11.1.0-6_amd64.deb](https://packages.ubuntu.com/jammy/libllvm11) for your architecture
- then you have to install [libphobos2-ldc-shared98_1.28.0-1ubuntu1_amd64.deb](https://packages.ubuntu.com/jammy/libphobos2-ldc-shared98) for your architecture
Then you can continue to install with your command `sudo apt install --no-install-recommends --no-install-suggests onedrive`
I've been able to configure the access to my tenant and the rest.
Thanks

**Reply by my-vegetable-has-exploded** [reply]
> That is what /etc/lsb-release says...
> 
> ```
> # more /etc/lsb-release
> DISTRIB_ID=Ubuntu
> DISTRIB_RELEASE=22.04
> DISTRIB_CODENAME=jammy
> DISTRIB_DESCRIPTION="Ubuntu 22.04.4 LTS"
> ```
> 
> Anyways downloading the deb packages from Ubuntu website and installing them with dpkg -i fixed the installation:
> 
>     * first you have to install [libllvm11_11.1.0-6_amd64.deb](https://packages.ubuntu.com/jammy/libllvm11) for your architecture
> 
>     * then you have to install [libphobos2-ldc-shared98_1.28.0-1ubuntu1_amd64.deb](https://packages.ubuntu.com/jammy/libphobos2-ldc-shared98) for your architecture
>       Then you can continue to install with your command `sudo apt install --no-install-recommends --no-install-suggests onedrive`
>       I've been able to configure the access to my tenant and the rest.
>       Thanks

Thanks for solution. For someone who needs libphobos2-ldc-shared94 in debian11, [bullseye/libphobos2-ldc-shared94.deb](https://packages.debian.org/bullseye/libphobos2-ldc-shared94) is your need.
